### PR TITLE
Migrate e2e tests to Cypress

### LINKED ISSUE #3921: Migrate e2e tests to Cypress

### PR DESCRIPTION
**Pull request by eloyekunle**
Fixes #3921

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,15 +24,15 @@
...
     "test": "npm run test:frontend && npm run test:backend && npm run test:e2e",
     "test:frontend": "ng test --karma-config ./aio/karma.conf.js --watch=false",
     "test:frontend:coverage": "ng test --karma-config ./aio/karma.conf.js --watch=false --code-coverage",
     "test:frontend:watch": "ng test --karma-config ./aio/karma.conf.js",
     "test:backend": "go test github.com/kubernetes/dashboard/src/app/backend/...",
     "test:backend:coverage": "./aio/scripts/coverage-backend.sh",
     "test:coverage": "npm run test:frontend:coverage && npm run test:backend:coverage",
-    "test:e2e": "ng e2e",
-    "pretest:e2e": "webdriver-manager update --standalone false --versions.chrome 2.45 --gecko false",
+    "test:e2e": "npx cypress run",
```
Comment (maciaszczykm) [inline package.json]: I'd change it to `e2e` so we can use `npm e2e` from now. `e2e` are not the standard test.

**Comment by eloyekunle** [comment]
@maciaszczykm PTAL.

**Comment by maciaszczykm** [comment]
Looks good so far, could you add example test? For example checking for buttons on login page - something very simple. Would be nice to add it to the CI and start running it.

Please also add a misssing newline at the end of the files. 

And remove not used files.

**Comment by eloyekunle** [comment]
I'm having problem figuring out why the cypress test is failing.

**Comment by eloyekunle** [comment]
So, apparently, the cypress test will fail when `aot` compilation is enabled, but succeeds otherwise.

@maciaszczykm I have added an example test to check sidebar collapse.

File: package.json [hunk 2]
```diff
@@ -9,7 +9,7 @@
   "scripts": {
     "start": "concurrently \"npm run start:backend\" \"npm run start:frontend --kubernetes-dashboard:bind_address=$npm_package_config_bind_address\"",
     "start:https": "concurrently \"npm run start:backend:https\" \"npm run start:frontend:https\"",
-    "start:frontend": "npm run postversion && ng serve --aot --progress=false --proxy-config aio/proxy.conf.json --host=$npm_package_config_bind_address",
```
Comment (maciaszczykm) [inline package.json]: @floreks We wanted to do it anyways, do I remember it correctly?

**Comment by maciaszczykm** [comment]
Please rebase.

/approve

File: cypress/integration/01-Startup.spec.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+describe("Application", () => {
```
Comment (floreks) [inline cypress/integration/01-Startup.spec.js]: Can we rename it to something else i.e. `basic.spec.js`?

File: cypress.json [hunk 4]
```diff
@@ -0,0 +1,6 @@
+{
+  "baseUrl": "http://localhost:8080",
+  "video": false,
+  "viewportWidth": 1000,
```
Comment (floreks) [inline cypress.json]: Those are quite strange viewport sizes. Can you use some standard resolutions?

**Comment by maciaszczykm** [comment]
Have you tested it locally after removing AOT compilation?

**Comment by eloyekunle** [comment]
@maciaszczykm Yes it works fine locally.

File: cypress/integration/01-Startup.spec.js (hunk 3 above)
Comment (eloyekunle) [reply cypress/integration/01-Startup.spec.js]: Yes, sure

File: cypress.json (hunk 4 above)
Comment (maciaszczykm) [reply cypress.json]: 1920x1080 would be the best IMO.

File: cypress/integration/01-Startup.spec.js (hunk 3 above)
Comment (eloyekunle) [reply cypress/integration/01-Startup.spec.js]: done

File: cypress.json (hunk 4 above)
Comment (eloyekunle) [reply cypress.json]: done. I'm using the Macbook 15 resolution (`1440x900`)

Comment (eloyekunle) [reply cypress.json]: Done.

**Comment by maciaszczykm** [comment]
Please regenerate package-lock.json as it's broken:

```
npm ERR! cipm can only install packages when your package.json and package-lock.json or npm-shrinkwrap.json are in sync. Please update your lock file with `npm install` before continuing.
npm ERR! 
npm ERR! 
npm ERR! Invalid: lock file's cypress@3.4.0 does not satisfy cypress@3.3.2
npm ERR! 

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/maciaszczykm/.npm/_logs/2019-07-11T13_03_15_733Z-debug.log

```

**Comment by maciaszczykm** [comment]
> @maciaszczykm Yes it works fine locally.

@eloyekunle It doesn't work for me. Router is broken as the lazy lading is the problem. Try to visit any non group state (i.e. pods, replica sets etc.).

Moreover, there are a lot of the errors in the console:

Zrzut ekranu 2019-07-11 o 15 28 53

Please verify it more closely.

**Comment by eloyekunle** [comment]
@maciaszczykm I'll look into it

**Comment by eloyekunle** [comment]
This is ready for a re-review.

**Comment by eloyekunle** [comment]
@KadenLNelson The main purpose of this PR is about replacing the test framework, not adding new tests, though this is also on the roadmap.

**Comment by maciaszczykm** [comment]
@eloyekunle Can you rebase? Is it ready besides that?

**Comment by eloyekunle** [comment]
Yes, it's ready. I've rebased.

**Comment by KadenLNelson** [comment]
@eloyekunle Ah, okay. Thank you for this! Is there an issue open for adding more tests?

**Comment by eloyekunle** [comment]
@KadenLNelson 

Yes, there are.
For unit tests, at #3865 
For e2e tests, at #4153

**Comment by anshulahuja98** [comment]
@eloyekunle I can't yet see any e2e test based on cypress in the repo/
I was thinking of working on some e2e tests, but I needed some reference tests to follow from...

**Pull request merged**